TypeSystem/README.md

```diff
@@ -78,16 +87,103 @@ CURRENT_STRUCT(Foo) {
   CURRENT_CONSTRUCTOR(Foo)(uint64_t i) : i(i) {}
 };
 ```
+ 
+### Templates
+
+Along with `CURRENT_STRUCT`, Current type system supports `CURRENT_STRUCT_T`: effectively a `template<typename T> struct ...`. The `CURRENT_CONSTRUCTOR_T` and `CURRENT_DEFAULT_CONSTRUCTOR_T` macros should be used instead of the non-`_T` ones.
+
+### Field Descriptions
+
+When a `CURRENT_STRUCT` defined a `CURRENT_FIELD(foo, ...);`, it may also define `CURRENT_FIELD_DESCRIPTION(foo, "This is the story of foo.");` Like fields, field descriptions can be reflected upon. Field descriptions can also be exported in various formats, most notably C++ itself (as the definition of the C++ struct), and (Markdown)[https://github.com/C5T/Current/blob/master/TypeSystem/Schema/golden/smoke_test_struct.md].
+
```

Comment:
Swap round and square brackets for the link? I.e. `(Markdown)[URL]` -> `[Markdown](URL)`.